hpx/util/unwrap_ref.hpp

```diff
@@ -0,0 +1,60 @@
+//  Copyright (c) 2017 Hartmut Kaiser
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+// hpxinspect:nodeprecatedinclude:boost/ref.hpp
+// hpxinspect:nodeprecatedname:boost::reference_wrapper
+
+#if !defined(HPX_UTIL_UNWRAP_REF_JAN_05_2017_0356PM)
+#define HPX_UTIL_UNWRAP_REF_JAN_05_2017_0356PM
+
+#include <hpx/config.hpp>
+
+#include <boost/ref.hpp>
+
+#include <functional>
+
+namespace hpx { namespace util
+{
+    template <typename T>
+    struct unwrap_reference
+    {
+        typedef T type;
+    };
+
+    template <typename T>
+    struct unwrap_reference<boost::reference_wrapper<T> >
+    {
+        typedef T type;
+    };
+
+    template <typename T>
+    struct unwrap_reference<boost::reference_wrapper<T> const>
+    {
+        typedef T type;
+    };
+
+    template <typename T>
+    struct unwrap_reference<std::reference_wrapper<T> >
+    {
+        typedef T type;
+    };
+
+    template <typename T>
+    struct unwrap_reference<std::reference_wrapper<T> const>
+    {
+        typedef T type;
+    };
+
+    template <typename T>
+    HPX_FORCEINLINE typename unwrap_reference<T>::type&
```

Comment:
As mentioned in an earlier comment. It would most likely rather confuse potential users of this trait. The original implementation in boost is done in the same manner. It was just a a thought on my end. Totally impartial here, as written previously, let's just keep it that way.